deepsearch/handler.py

```diff
@@ -0,0 +1,28 @@
+import json
+import os
+
+from notebook.base.handlers import APIHandler
+
+
+class DeepsearchHandler(APIHandler):
+    def get(self, **kwargs):
+        quarry = self.get_argument('quarry')
+        # TODO: consider validating the path. don't allow dir traversal.
+        directory = self.settings['contents_manager'].root_dir + "\\" + self.get_argument('dir')
+        self.finish(json.dumps(self.search_in_files(quarry, directory)))
+
+    @staticmethod
+    def search_in_files(quarry, directory):
+        results = []
+        for folder, dirs, files in os.walk(directory):
+            for file in files:
+                # TODO: Consider checking for file type (can use mimetypes.guess_type())
```

Comment:
Actually, this is extremely important.
We would not like to search on `ipynb_cache` or anything else, but `*.py`, `*.ipynb`, `*.md`

Reply:
what about .txt and other types we didn't thought about? 
maybe it's better to use a black list rather than a white list?